src/ucp/core/ucp_ep.c

```diff
@@ -212,12 +219,23 @@ ucs_status_t ucp_ep_create(ucp_worker_h worker,
         goto err_destroy_ep;
     }
 
+    /* Setup error handler */
```

Comment:
missed it in previous PR, but there is a problem with ucp_ep_config(ep)->err_mode. Need to configure the err mode inside the "key" field of ep_config **before** filling the rest of the configuration. Consider 2 endpoints which have same configuration, except err handling. they will end up using same config struct and override each other settings.